### PR TITLE
Added the styles for the new gutenberg-based search widget

### PR DESCRIPTION
**Pull request by kirannasim**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I pulled my branch from `develop`.
- [x] I am submitting my pull request to `develop`.
- [x] I have resolved any conflicts merging this pull request would create.
- [x] I have checked that there aren't other open Pull Requests for the same update/change.
- [x] My code follows the code style of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] \(Optional) My change requires a change to the documentation.
- [ ] \(Optional) I have updated the documentation accordingly.
- [ ] \(Optional) My change requires a change to the translations.
- [ ] \(Optional) I have updated the translations accordingly.
- [ ] `composer cs:check` has passed locally.
- [ ] `composer lint:php` has passed locally.
- [x] I have read the **[CONTRIBUTING](https://github.com/understrap/understrap/blob/main/.github/CONTRIBUTING.md)** document.

### CODE REVIEW

**Review by bacoords — APPROVED** [review]

File: src/sass/theme/_understrap.scss [hunk 1]
```diff
@@ -170,6 +170,41 @@ a.skip-link {
...
+        .wp-block-search__button {
+            border-radius: $border-radius;
+
+            &:not(:first-child):not(.dropdown-menu):not(.valid-tooltip):not(.valid-feedback):not(.invalid-tooltip):not(.invalid-feedback) {
+                margin-left: 0;
+            }
+
+			@if variable-exists('bootstrap4') {
+				@include button-variant( $primary, $primary );
+				border-top-left-radius: 0px;
```
Comment (IanDelMar) [inline src/sass/theme/_understrap.scss:199]:
```diff
- border-top-left-radius: 0px;
+ border-top-left-radius: 0;
```

File: src/sass/theme/_understrap.scss [hunk 2]
```diff
@@ -170,6 +170,41 @@ a.skip-link {
...
+				border-bottom-left-radius: 0px;
```
Comment (IanDelMar) [inline src/sass/theme/_understrap.scss:200]:
```diff
- border-bottom-left-radius: 0px;
+ border-bottom-left-radius: 0;
```

File: src/sass/theme/_understrap.scss [hunk 3]
```diff
@@ -170,6 +170,41 @@ a.skip-link {
 	display: block;
 }
 
+// New Search Widget in Block Editor
+.wp-block-search {
+    .wp-block-search__label {
+		@if variable-exists('bootstrap4') {
+			@include sr-only();
```
Comment (IanDelMar) [inline src/sass/theme/_understrap.scss:177]: I think @extend would give smaller CSS.

File: src/sass/theme/_understrap.scss [hunk 4]
```diff
@@ -170,6 +170,41 @@ a.skip-link {
...
+		} @else {
+			@include visually-hidden();
```
Comment (IanDelMar) [inline src/sass/theme/_understrap.scss:179]: I think @extend would give smaller CSS.

File: src/sass/theme/_understrap.scss [hunk 5]
```diff
@@ -170,3 +170,28 @@ a.skip-link {
...
+.wp-block-search {
+    .wp-block-search__label {
+        @extend .screen-reader-text;
+    }
+
+    .wp-block-search__inside-wrapper {
+        @extend .input-group;
+
+        .wp-block-search__input {
+            @extend .form-control;
```
Comment (IanDelMar) [inline src/sass/theme/_understrap.scss:187]: @bacoords a from-control mixin as mentioned in #1661 would make perfectly sense here.

File: src/sass/theme/_understrap.scss [hunk 6]
```diff
@@ -190,10 +194,17 @@ a.skip-link {
                 margin-left: 0;
             }
 
-            @extend .btn-primary;
+			@if variable-exists('bootstrap4') {
+				@include button-variant( $primary, $primary );
```
Comment (IanDelMar) [inline src/sass/theme/_understrap.scss:198]: `@extend .btn-primary;` gives marginally smaller CSS

File: src/sass/theme/_understrap.scss [hunk 7]
```diff
@@ -170,6 +170,41 @@ a.skip-link {
...
+        }
+
+        .wp-block-search__button {
+            border-radius: $border-radius;
+
+            &:not(:first-child):not(.dropdown-menu):not(.valid-tooltip):not(.valid-feedback):not(.invalid-tooltip):not(.invalid-feedback) {
+                margin-left: 0;
+            }
+
+			@if variable-exists('bootstrap4') {
```
Comment (IanDelMar) [inline src/sass/theme/_understrap.scss:197]: Can be simplified to
```scss
@include button-variant( $primary, $primary );
@if variable-exists('bootstrap4') {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}
```

**Pull request merged**